akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/FileUploadExamplesSpec.scala

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package docs.http.scaladsl.server
+
+import java.io.File
+
+import akka.http.scaladsl.model.Multipart.FormData.BodyPart
+import akka.stream.io.SynchronousFileSink
+import akka.stream.scaladsl._
+import akka.http.scaladsl.model.Multipart
+
+import scala.concurrent.Future
+
+class FileUploadExamplesSpec extends RoutingSpec {
+
+  //#simple-upload
+  val simpleRoute =
+    path("upload") {
+      entity(as[Multipart.FormData]) { formData =>
+        val completed: Source[File, _] = formData.parts.mapAsync(1) {
+          case BodyPart(name, entity, _, _) =>
+            val file = File.createTempFile("uploaded", "bin")
+            val sink = SynchronousFileSink(file)
+            entity.dataBytes.runWith(sink).map(_ => file)
```

Comment:
I thought it simpler to show a csv here than to go all in with json parsing (which needs implicit context), for the sample we can just `via(framing).mapConcat(_.split(","))`